jetty-util/src/main/java/org/eclipse/jetty/util/Callback.java

```diff
@@ -210,6 +210,58 @@ public void failed(Throwable x)
         };
     }
 
+    /**
+     * Create a nested callback which always fails the nested callback on completion.
+     * @param callback The nested callback
+     * @param cause The cause to fail the nested callback, if the new callback is failed the reason
+     *             will be added to this cause as a suppressed exception.
+     * @return a new callback.
+     */
+    static Callback from(Callback callback, Throwable cause)
+    {
+        return new Callback()
+        {
+            @Override
+            public void succeeded()
+            {
+                callback.failed(cause);
+            }
+
+            @Override
+            public void failed(Throwable x)
+            {
+                cause.addSuppressed(x);
+                callback.failed(cause);
+            }
+        };
+    }
+
+    /**
+     * Create a callback which combines two other callbacks and will succeed or fail them both.
+     * @param callback1 The first callback
+     * @param callback2 The second callback
+     * @return a new callback.
+     */
+    static Callback from(Callback callback1, Callback callback2)
+    {
+        return new Callback()
+        {
+            @Override
+            public void succeeded()
+            {
+                callback1.succeeded();
```

Comment:
@sbordet the from style is something we've been using in websockets.  They are not used elsewhere because most of the code was written pre-lambda.   While we might perhaps have too many from methods now, I do think we should at least consider if they are utility and are usable elsewhere.

Reply:
the (Callback, Throwable) version is used 5 places in `WebSocketProxy` and also 2 places in `WebSocketChannel`.
the (Callback, Callback) one is used 4 times and only in `WebSocketProxy`